src/vs/workbench/parts/debug/electron-browser/debugHover.ts

```diff
@@ -112,26 +112,67 @@ export class DebugHoverWidget implements editorbrowser.IContentWidget {
 			return;
 		}
 
-		// string magic to get the parents of the variable (a and b for a.b.foo)
 		const lineContent = model.getLineContent(pos.lineNumber);
-		const namesToFind = lineContent.substring(0, lineContent.indexOf('.' + hoveringOver))
-			.split('.').map(word => word.trim()).filter(word => !!word);
-		namesToFind.push(hoveringOver);
-		namesToFind[0] = namesToFind[0].substring(namesToFind[0].lastIndexOf(' ') + 1);
+		const session = this.debugService.getActiveSession();
+		let evaluatedExpression : TPromise<Expression> = undefined;
+		let matchingExpression: string = undefined;
+		let startOffset: number = 0;
+		if (session.configuration.capabilities.supportsEvaluateForHovers) {
+			let expression: RegExp = /([^()\[\]{}<>\s+\-/%~#^;=|,`!]|\->)+/g;
+			let result: RegExpExecArray = undefined;
+
+			while (result = expression.exec(lineContent)) {
+				let start = result.index + 1;
+				let end = start + result[0].length;
+
+				if (start <= range.startColumn && end >= range.endColumn) {
+					matchingExpression = result[0];
+					startOffset = start;
+					break;
+				}
+			}
 
-		return this.getExpression(namesToFind).then(expression => {
+			if (matchingExpression) {
+				let subExpression: RegExp = /\w+/g;
+				let subExpressionResult: RegExpExecArray = undefined;
+				while (subExpressionResult = subExpression.exec(matchingExpression)) {
+					let subStart = subExpressionResult.index + 1 + startOffset;
+					let subEnd = subStart + subExpressionResult[0].length;
+					if (subStart <= range.startColumn && subEnd >= range.endColumn) {
```

Comment:
Agreed. (Copy paste error) 

---
In reply to: [72359397](https://github.com/Microsoft/vscode/pull/9821#discussion_r72359397) [](ancestors = 72359397)